File: app/src/main/java/it/paranoidsquirrels/idleguildmaster/storage/data/items/instances/InsectEgg.java
package it.paranoidsquirrels.idleguildmaster.storage.data.items.instances;

import it.paranoidsquirrels.idleguildmaster.R;
import it.paranoidsquirrels.idleguildmaster.Utils;
import it.paranoidsquirrels.idleguildmaster.storage.data.pets.Pet;

/* JADX INFO: loaded from: classes3.dex */
public class InsectEgg extends it.paranoidsquirrels.idleguildmaster.storage.data.items.abstractClasses.Egg {
    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.items.abstractClasses.Consumable, it.paranoidsquirrels.idleguildmaster.storage.data.items.Item
    protected void configureProperties() {
        this.idName = R.string.egg_insect_name;
        this.idDescription = R.string.egg_insect_description;
        this.idImage = R.drawable.egg_insect;
        this.source.add(Integer.valueOf(R.string.dungeon_name_the_desert));
        this.source.add(Integer.valueOf(R.string.dungeon_name_obsidian_mines));
        this.source.add(Integer.valueOf(R.string.dungeon_name_the_southern_grove));
        this.price = 100L;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.items.abstractClasses.Egg
    public Pet hatch() {
        int iCalculateNewPetId = Utils.calculateNewPetId();
        double dRandom = Utils.random();
        if (dRandom < 0.75d) {
            return Pet.getInstance("Mosquito", iCalculateNewPetId);
        }
        return dRandom < 0.95d ? Pet.getInstance("Beetle", iCalculateNewPetId) : Pet.getInstance("Tarantula", iCalculateNewPetId);
    }
}

File: app/src/main/kotlin/it/paranoidsquirrels/idleguildmaster/storage/data/items/instances/InsectEgg.kt
package it.paranoidsquirrels.idleguildmaster.storage.data.items.instances

import it.paranoidsquirrels.idleguildmaster.R
import it.paranoidsquirrels.idleguildmaster.Utils
import it.paranoidsquirrels.idleguildmaster.storage.data.items.abstractClasses.Egg
import it.paranoidsquirrels.idleguildmaster.storage.data.pets.Pet

class InsectEgg : Egg() {
    override fun configureProperties() {
        idName = R.string.egg_insect_name
        idDescription = R.string.egg_insect_description
        idImage = R.drawable.egg_insect
        source.add(R.string.dungeon_name_the_desert)
        source.add(R.string.dungeon_name_obsidian_mines)
        source.add(R.string.dungeon_name_the_southern_grove)
        price = 100L
    }

    override fun hatch(): Pet? {
        val id = Utils.calculateNewPetId()
        val r = Utils.random()
        return if (r < 0.75) {
            Pet.getInstance("Mosquito", id)
        } else if (r < 0.95) {
            Pet.getInstance("Beetle", id)
        } else {
            Pet.getInstance("Tarantula", id)
        }
    }
}
